src/Idris/Parser.hs

```diff
@@ -711,9 +712,10 @@ postulate syn = do (doc, ext)
                                    ext <- ppostDecl
                                    return (doc, ext)
                    ist <- get
-                   let initOpts = if default_total ist
-                                     then [TotalFn]
-                                     else []
+                   let initOpts = case default_total ist of
+                                    DefaultCheckingTotal    -> [TotalFn]
+                                    DefaultCheckingCovering -> [CoveringFn]
+                                    _                       -> []
                    opts <- fnOpts initOpts
```

Comment:
So it would be something like:

```
...
opts <- fnOpts [] -- Ultimately the parameter should be dropped, but I am unsure how for now
acc <- accesbility
opts' <- fnOpts opts
let existingTotality = opts' `intersect` [TotalFn, CoveringFn, PartialFn]
opts'' <- case opts' of
  [] -> 
     case default_total ist of
        DefaultCheckingTotal -> return (TotalFn : opts')
        ... -- The rest of the cases
  [x] -> -- already has one
  _ -> -- has more than one, fail or provide an error message
... -- use opts'' instead of opts' later
```